### PR TITLE
Typescript migration

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "ts": "yarn run test-simple",
     "tsv": "yarn run test-simple --verbose",
     "u": "yarn upgrade --latest --force",
-    "i": "yarn install --frozen-lockfile"
+    "i": "yarn install --frozen-lockfile",
+    "ts-test": "nyc mocha \"./test/**/*.spec.ts\""
```
Comment (MacMcIrish) [inline package.json]: Needs to be in robo-config-plugin

**Comment by simlu** [comment]
ping :)

**Comment by yurist38** [comment]
Hey! I just wanted to use `object-scan` in my TS project and realized that there are no TS typings yet, such a bummer... Do you still consider the migration? I'd like to assist if needed.

**Comment by simlu** [comment]
I was actually just thinking about this today as I was working on this project. Yes, we're still planning to migrate to typescript. Unfortunately not sure when I'm going to get to it.

I'll let you know when the "busy work" is done and we're getting to the translation. I don't have a lot of experience with TS, so help might be very welcome at that point!

**Comment by yurist38** [comment]
Nice! Moving to TS is always a great idea! 😄  Feel free to contact me, I'd love to help...

**Comment by simlu** [comment]
@yurist38 Was playing around with it, but couldn't get it to work to my satisfaction. Do you have any experience with using babel + typescript by any chance?

**Comment by yurist38** [comment]
@simlu Yes, I do. Used Babel with TS for a while on my previous job. What is the problem? Can I reproduce it on this branch?

**Comment by simlu** [comment]
@yurist38 I'll put up a wip pr later for a different repo that has fewer files and then we can discuss. Cheers

**Comment by simlu** [comment]
@yurist38 PR is up here: https://github.com/blackflux/object-treeify/pull/841/files

Closing this pr for now

**Pull request closed without merging**